### DISCUSSION TITLE
Local plugin doesn't appear in admin.

### DISCUSSION THREAD

**Discussion by augustbjornberg** [strapi/strapi#6838] [Legacy Topics]
I'm reading the little official documentation there is on custom plugins, followed [this](https://strapi.io/blog/how-to-create-an-import-content-plugin-part-1-4) tutorial, and a few others. But no matter what i do, no local plugin ever appears in the admin.

I've created a test plugin with `strapi generate:plugin test`. The plugin is generated successfully with all files needed, but after launching the admin, i can't find the plugin in the left hand menu, or navigate to it in any way. The plugin is visible in the Roles & Permissions tab, but nowhere else.

What do i have to do to make the plugin appear in the admin?

**Comment by MattieBelt** [comment]
Hey @augustbjornberg did you look at this section of the front-end plugin guide:
https://strapi.io/documentation/v3.x/plugin-development/frontend-development.html#main-plugin-object

**Reply by augustbjornberg** [reply]
Hey, I have. I couldn't immediately see anything that would cause the component to not render. I've switched a few props to see if anything would happen, but with no difference. Maybe i've missed something.

I was under the assumption that plugins made via the command line would be visible by default, is this not the case? 

The documentation on plugins feels very limited for the uninitiated, there's even properties made by the plugin creation command that aren't mentioned in the documentation. I wish it was more in depth.

./\<plugin name\>/admin/src/index.js

    export default strapi => {
        const pluginDescription = pluginPkg.strapi.description || pluginPkg.description;

        const plugin = {
            blockerComponent: null,
            blockerComponentProps: {},
            description: pluginDescription,
            icon: pluginPkg.strapi.icon,
            id: pluginId,
            initializer: Initializer,
            injectedComponents: [],
            isReady: false,
            isRequired: pluginPkg.strapi.required || false,
            layout: null,
            lifecycles,
            leftMenuLinks: [],
            leftMenuSections: [],
            mainComponent: App,
            name: pluginPkg.strapi.name,
            preventComponentRendering: false,
            trads,
        };

        return strapi.registerPlugin(plugin);
    };

**Reply by soupette** [reply]
You need to start your app in development mode in order to see your local changes. If you run `strapi build` you should be able to see your plugin in the left menu when you are on the `1337` port. If you want to customise the front-end files you need to start your app with `yarn develop --watch-admin`

**Reply by augustbjornberg** [reply]
Wow, i was blind. I completely missed `--watch-admin` earlier. Thank you @soupette . 

`strapi build` did the job, I'm able see the plugin now. Unfortunately `--watch-admin` shows me a spinner that goes on for eternity on every page, no errors as far as I can tell. Seems unrelated, but do you have a hunch what might cause this?

**Reply by soupette** [reply]
Can you copy/paste your `plugin/admin/src/index.js` file you might be missing the `isReady` property so the app loads until your plugin emits the event that it is ready.

**Reply by augustbjornberg** [reply]
**EDIT at the bottom.**

Absolutely, here you go. I was thinking it might have something to do with the `isReady` property too, but i wasn't able to figure out if that was the cause. To debug, I tried simply setting `isReady` to true, but that didn't do the trick.

`plugin/admin/src/index.js`, note that I've made no changes since generating the plugin.

	import pluginPkg from '../../package.json';
	import pluginId from './pluginId';
	import App from './containers/App';
	import Initializer from './containers/Initializer';
	import lifecycles from './lifecycles';
	import trads from './translations';

	export default strapi => {
		const pluginDescription = pluginPkg.strapi.description || pluginPkg.description;

		const plugin = {
			blockerComponent: null,
			blockerComponentProps: {},
			description: pluginDescription,
			icon: pluginPkg.strapi.icon,
			id: pluginId,
			initializer: Initializer,
			injectedComponents: [],
			isReady: false,
			isRequired: pluginPkg.strapi.required || false,
			layout: null,
			lifecycles,
			leftMenuLinks: [],
			leftMenuSections: [],
			mainComponent: App,
			name: pluginPkg.strapi.name,
			preventComponentRendering: false,
			trads,
		};

		return strapi.registerPlugin(plugin);
	};

It seems like `plugin/admin/src/Initializer/index.js` is handles the effect of setting the property to true. This isn't how i normally handle properties between instances in react, so i'm not sure if it's working or not. Also no changes made since generating the plugin.

	import { useEffect, useRef } from 'react';
	import PropTypes from 'prop-types';
	import pluginId from '../../pluginId';

	const Initializer = ({ updatePlugin }) => {
		const ref = useRef();
		ref.current = updatePlugin;

		useEffect(() => {
			ref.current(pluginId, 'isReady', true);
		}, []);

		return null;
	};

	Initializer.propTypes = {
		updatePlugin: PropTypes.func.isRequired,
	};

	export default Initializer;

**EDIT:** 

I removed the plugin completely, and the admin is still stuck on the spinner when using `--watch-admin`. Does that confirm that this issue is unrelated to the plugin? 

How do i best debug this further?
